src/module/functions/mod.rs

```diff
@@ -418,6 +433,66 @@ impl Module {
 
         Ok(())
     }
+
+    /// Retrieve an exported function by name
+    pub fn get_exported_func_by_name(&self, name: impl AsRef<str>) -> Result<FunctionId> {
+        self.exports
+            .iter()
+            // Find the export with the correct name and internal type
+            .filter_map(|expt| match expt.item {
+                ExportItem::Function(fid) if expt.name == name.as_ref() => Some(fid),
+                _ => None,
+            })
+            .nth(0)
+            .with_context(|| format!("unable to find function export '{}'", name.as_ref()))
+    }
+
+    /// Retrieve an imported function by name
+    pub fn get_imported_func_by_name(&self, name: impl AsRef<str>) -> Result<FunctionId> {
+        self.imports
+            .iter()
+            // Find the export with the correct name and internal type
+            .filter_map(|impt| match impt.kind {
+                ImportKind::Function(fid) if impt.name == name.as_ref() => Some(fid),
+                _ => None,
+            })
+            .nth(0)
+            .with_context(|| format!("unable to find function export '{}'", name.as_ref()))
+    }
+
+    /// Retrieve the ID for the first exported memory.
+    ///
+    /// This method does not work in contexts with [multi-memory enabled](https://github.com/WebAssembly/multi-memory),
+    /// and will error if more than one memory is present.
+    pub fn get_primary_memory_id(&self) -> Result<MemoryId> {
+        if self.memories.len() > 1 {
+            bail!("multiple memories unsupported")
+        }
+
+        self.memories
+            .iter()
+            .next()
+            .map(Memory::id)
+            .context("module does not export a memory")
+    }
+
+    /// Replace a single function with the result
+    ///
+    /// When called, if `builder` produces a None value, the function in question will be
+    /// replaced with a stub that does nothing (more precisely, a function with an unreachable body).
+    pub fn replace_fn_by_name<F>(&mut self, name: impl AsRef<str>, builder: F) -> Result<()>
+    where
+        F: FnOnce() -> Result<Option<Function>>,
+    {
+        let built_fn = builder().context("fn builder failed")?;
+
+        // TODO: Hmnn.... how should we distinguish between which one to go for?
```

Comment:
I'd suggest having this function take a function id instead to resolve this.

Reply:
Will do, that's a good way to make it explicit.

- [ ] Take ID for replacing function